tokio/src/runtime/queue.rs

```diff
@@ -324,83 +341,114 @@ impl<T> Steal<T> {
         }
 
         // Synchronize with stealers
-        let dst_head = dst.inner.head.load(Acquire);
-
-        assert!(dst_tail.wrapping_sub(dst_head) + n <= LOCAL_QUEUE_CAPACITY as u32);
+        let (dst_steal, dst_real) = split(dst.inner.head.load(Acquire));
+        assert_eq!(dst_steal, dst_real);
 
         // Make the stolen items available to consumers
         dst.inner.tail.store(dst_tail.wrapping_add(n), Release);
 
         Some(ret)
     }
 
-    fn steal_into2(&self, dst: &mut Local<T>, dst_tail: u32) -> u32 {
-        loop {
-            let src_head = self.0.head.load(Acquire);
+    fn steal_into2(&self, dst: &mut Local<T>, dst_tail: u8) -> u8 {
```

Comment:
It would be nice if there was a comment explaining what this does and what it's returning. But, there wasn't previously, so not a blocker.